storage/txcache/monitoring.go

```diff
@@ -42,10 +42,47 @@ func (cache *TxCache) monitorSelectionStart() *core.StopWatch {
 func (cache *TxCache) monitorSelectionEnd(selection []*WrappedTransaction, stopWatch *core.StopWatch) {
 	stopWatch.Stop("selection")
 	duration := stopWatch.GetMeasurement("selection")
-	log.Debug("TxCache: selection ended", "name", cache.name, "duration", duration, "numTxSelected", len(selection))
+	numSendersSelected := cache.numSendersSelected.Reset()
+	numSendersWithInitialGap := cache.numSendersWithInitialGap.Reset()
+	numSendersWithMiddleGap := cache.numSendersWithMiddleGap.Reset()
+	numSendersInGracePeriod := cache.numSendersInGracePeriod.Reset()
+
+	log.Debug("TxCache: selection ended", "name", cache.name, "duration", duration,
+		"numTxSelected", len(selection),
+		"numSendersSelected", numSendersSelected,
+		"numSendersWithInitialGap", numSendersWithInitialGap,
+		"numSendersWithMiddleGap", numSendersWithMiddleGap,
+		"numSendersInGracePeriod", numSendersInGracePeriod,
+	)
+
 	cache.displaySendersHistogram()
 }
 
+type batchSelectionJournal struct {
+	isFirstBatch  bool
+	copied        int
+	hasInitialGap bool
+	hasMiddleGap  bool
+	isGracePeriod bool
+}
+
+func (cache *TxCache) monitorBatchSelectionEnd(journal batchSelectionJournal) {
+	if journal.isFirstBatch {
```

Comment:
Can you please refactor this? nested else-ifs inside an if? Why not following the pattern we use in all the rest of the project: if not_ok, return

Reply:
Missed that, fixed :)